install_config/registry/deploy_registry_existing_clusters.adoc

```diff
@@ -91,7 +94,147 @@ Daemonsets ensure that when nodes are created, they contain copies of a
 specified pod. When the nodes are removed, the pods are garbage collected.
 
 For more information on `DaemonSets`, see
-xref:../../dev_guide/daemonsets.adoc#dev-guide-daemonsets[Using Daemonsets].
+xref:../../dev_guide/daemonsets.adoc#dev-guide-daemonsets[Using DaemonSets].
+
+[[redeploy-registry]]
+=== Re-deploying the Registry
+
+There is a couple of things to keep on mind when re-deploying an existing docker-registry
+DeploymentConfig or DaemonSet:
+
+1. Service IP shall be preserved.
+2. Any change to the service IP is not propagated to the {product-title} cluster's components
+until the master API and controllers are restarted.
+3. The docker-registry service must exist before the docker-registry pod is created.
+
+**Service IP shall be preserved** because master caches the docker-registry's internal address.
+Unless one of the following applies, the service IP and port will be used to construct docker
+image pull-spec of the newly pushed images.
+
+- `OPENSHIFT_DEFAULT_REGISTRY` environment variable not containing
+  the service IP is passed to the master process 
+- xref:extended_registry_configuration.adoc#setting-the-registry-hostname[
+  `imagePolicyConfig:internalRegistryHostname`] not containing service IP is set in the master
+  configuration file
+
+It's important to keep the pull-spec's registry part of managed images and image stream
+repositories the same for all the images in the cluster because some tools (e.g. `oc adm prune
+images`) determine the address of the docker registry from the existing images unless directly
+provided.
+
+It's recommended to use the DNS registry name instead of the service address in any case. But even
+though used, preserving the registry's service IP will ensure consistency with whatever
+image-related resource already exists in the cluster which will make possible future migration
+easier.
+
+Before reading further, keep in mind also the following recommendations:
+
+1. Back up the existing configuration.
+2. It's usually safer and faster to modify the existing working deployment than starting from a
+scratch.
+3. When the above is satisfied, you can return back to the earlier working version with
+`oc rollout undo ...`
+
+[NOTE]
+====
+Unfortunately the `oc rollout` is currently limited just to the DeploymentConfig. Do the backup
+for the DaemonSet installation.
```

Comment:
@tnozicka Any change regarding this for 3.10/3.11?

Reply:
I think `roullout` is supported since like Kubernetes 1.7

Reply:
don't say `unfortunately`, we apologize for nothing! :)  (but seriously, it strikes an awkward tone. just state the facts).

Reply:
> I think roullout is supported since like Kubernetes 1.7

Well, this is what I get with `oc v3.9.16-1+23bf34b-dirty`:

```
$ oc rollout latest ds/docker-registry
error: docker-registry is not a deployment config
```

> don't say unfortunately

My disappontment leaked into the text. I'll try to be less emotional ;-).

Reply:
You said `oc rollout`, not `oc rollout latest`. DS (with rolling Update strategy set) rollout automatically on config change. You can do e.g. `oc rollout undo daemonset <daemonset-name> --to-revision=<revision>` to rollback. If you need to force redeployment without a change in PodSpec you need to e.g. annotate the pod template with current time until we implement `rolling-restart`.